### PR TITLE
Expand MV doc around grace period

### PR DESCRIPTION
**Pull request by mosabua**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/iceberg.md [hunk 1]
```diff
@@ -1450,13 +1450,20 @@ it is being refreshed. Refreshing a materialized view also stores the
...
-and the data is queried directly from the base tables. Detecting outdated data
-is possible only when the materialized view uses Iceberg tables only, or when it
-uses a mix of Iceberg and non-Iceberg tables but some Iceberg tables are outdated.
-When the materialized view is based on non-Iceberg tables, querying it can
-return outdated data, since the connector has no information whether the
-underlying non-Iceberg tables have changed.
+date.
+
+The grace period of a materialized view defaults to infinity, and after each
+refresh the materialized view therefore behaves like a static table.
```
Comment (colebow) [inline docs/src/main/sphinx/connector/iceberg.md:1456]:
```suggestion
refresh with an infinite grace period, the materialized view then
behaves like a static table.
```
Might also be worth explicitly calling out

File: docs/src/main/sphinx/sql/create-materialized-view.md [hunk 2]
```diff
@@ -36,10 +36,11 @@ Note that `OR REPLACE` and `IF NOT EXISTS` are mutually exclusive clauses.
 
 The optional `GRACE PERIOD` clause specifies how long the query materialization
 is used for querying. If the time elapsed since last materialized view refresh
-is greater than the grace period, the materialized view acts as a normal view and
-the materialized data is not used. If not specified, the grace period defaults to
-infinity. See {doc}`refresh-materialized-view` for more about refreshing
-materialized views.
+is greater than the grace period, the materialized view acts as a normal view
+and the materialized data is not used. If not specified, the grace period
+defaults to infinity, and after each refresh the materialized view therefore
```
Comment (colebow) [inline docs/src/main/sphinx/sql/create-materialized-view.md:41]:
```suggestion
defaults to infinity, and after each refresh, the materialized view then
```

**Review by colebow — CHANGES_REQUESTED** [review]
Just a couple wording tweaks

File: docs/src/main/sphinx/connector/iceberg.md [hunk 3]
```diff
@@ -1450,13 +1450,20 @@ it is being refreshed. Refreshing a materialized view also stores the
...
+
+If the data is outdated and the grace period of the materialized view is
+elapsed, the materialized view behaves like a normal view, and the data is
+queried directly from the base tables.
+
+Detecting outdated data is possible only when the materialized view uses Iceberg
+tables only, or when it uses a mix of Iceberg and non-Iceberg tables but some
+Iceberg tables are outdated. When the materialized view is based on non-Iceberg
+tables, querying it can return outdated data, since the connector has no
+information whether the underlying non-Iceberg tables have changed.
```
Comment (findepi) [inline docs/src/main/sphinx/connector/iceberg.md:1466]: When there are non-Iceberg tables, view is considered stale once GRACE elapses.

Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md]: Tried to pull this in with more rewording.

**Comment by mosabua** [comment]
Rejigged this all some more @findepi @colebow

File: docs/src/main/sphinx/connector/iceberg.md [hunk 4]
```diff
@@ -1450,13 +1450,18 @@ it is being refreshed. Refreshing a materialized view also stores the
...
-When the materialized view is based on non-Iceberg tables, querying it can
-return outdated data, since the connector has no information whether the
-underlying non-Iceberg tables have changed.
+date.
+
+The grace period of a materialized view defaults to infinity. After each refresh
+with this infinite grace period, the materialized view then behaves like a
+static table with the data from the time of the refresh.
+
+If the data is outdated and the grace period of the materialized view is
```
Comment (colebow) [inline docs/src/main/sphinx/connector/iceberg.md:1459]:
```suggestion
If the data is outdated and the grace period of the materialized view has
```

**Review by colebow — APPROVED** [review]
One final nit then LGTM

**Review by findepi — APPROVED** [review]
skimmed, lgtm.

@martint may want to take a look

**Comment by mosabua** [comment]
Can you review again @colebow and @findepi  .. I adjusted quite a bit after I chatted with @martint

**Comment by mosabua** [comment]
Ping @colebow @findepi and @martint

File: docs/src/main/sphinx/connector/iceberg.md [hunk 5]
```diff
@@ -1463,13 +1463,12 @@ it is being refreshed. Refreshing a materialized view also stores the
...
-and the data is queried directly from the base tables. Detecting outdated data
-is possible only when the materialized view uses Iceberg tables only, or when it
-uses a mix of Iceberg and non-Iceberg tables but some Iceberg tables are outdated.
-When the materialized view is based on non-Iceberg tables, querying it can
-return outdated data, since the connector has no information whether the
-underlying non-Iceberg tables have changed.
+date.
+
+Materialized views that use non-Iceberg tables in the query show the [default
+behavior around grace period](mv-grace-period). If all tables are Iceberg
```
Comment (colebow) [inline docs/src/main/sphinx/connector/iceberg.md:1469]:
```suggestion
behavior around grace periods](mv-grace-period). If all tables are Iceberg
```

File: docs/src/main/sphinx/sql/create-materialized-view.md [hunk 6]
```diff
@@ -34,12 +34,21 @@ created if it does not exist yet.
...
-is greater than the grace period, the materialized view acts as a normal view and
-the materialized data is not used. If not specified, the grace period defaults to
-infinity. See {doc}`refresh-materialized-view` for more about refreshing
-materialized views.
+is used for querying:
+
+*  Within the grace period since last refresh, data retrieval is highly
+   performant because the query materialization is used. However, the data may not
+   be up to date with the base tables.
+* After the grace period elapsed, the data of the materialized view is computed
```
Comment (colebow) [inline docs/src/main/sphinx/sql/create-materialized-view.md:45]:
```suggestion
* After the grace period has elapsed, the data of the materialized view is computed
```

File: docs/src/main/sphinx/sql/create-materialized-view.md [hunk 7]
```diff
@@ -34,12 +34,21 @@ created if it does not exist yet.
...
+
 The optional `GRACE PERIOD` clause specifies how long the query materialization
-is used for querying. If the time elapsed since last materialized view refresh
-is greater than the grace period, the materialized view acts as a normal view and
-the materialized data is not used. If not specified, the grace period defaults to
-infinity. See {doc}`refresh-materialized-view` for more about refreshing
-materialized views.
+is used for querying:
+
+*  Within the grace period since last refresh, data retrieval is highly
```
Comment (colebow) [inline docs/src/main/sphinx/sql/create-materialized-view.md:42]: Spacing here is inconsistent with the lower bullet points.

**Review by colebow — APPROVED** [review]

**Comment by mosabua** [comment]
I did another round of checks and I think this is good to go.

**Pull request merged**